demo/src/main/java/com/google/android/fhir/demo/AddPatientViewModel.kt

```diff
@@ -53,23 +54,26 @@ class AddPatientViewModel(application: Application, private val state: SavedStat
    */
   fun savePatient(questionnaireResponse: QuestionnaireResponse) {
     viewModelScope.launch {
+      QuestionnaireResponseValidator.validateQuestionnaireResponse(
+          questionnaireResource,
+          questionnaireResponse,
+          getApplication()
+        )
+        .values
+        .flatten()
+        .filterNot { it.isValid }
+        .let {
+          if (it.isNotEmpty()) {
+            isPatientSaved.value = false
+            return@launch
+          }
+        }
```

Comment:
```suggestion
        .any { !it.isValid }
```
and wrap this in an `if` statement to return